Library/Formula/hbase.rb

```diff
@@ -4,32 +4,134 @@ class Hbase < Formula
+                  <value>#{var}/zookeeper</value>
+                </property>
+                <property>
+                  <name>hbase.zookeeper.dns.interface</name>
+                  <value>lo0</value>
+                </property>
+                <property>
+                  <name>hbase.regionserver.dns.interface</name>
+                  <value>lo0</value>
+                </property>
+                <property>
+                  <name>hbase.master.dns.interface</name>
+                  <value>lo0</value>
+                </property>
+              EOS
+
+    (var/"run/hbase").mkpath
   end
 
-  def caveats; <<-EOS.undent
-    You must edit the configs in:
-      #{libexec}/conf
-    to reflect your environment.
+  plist_options :manual => "#{HOMEBREW_PREFIX}/opt/hbase/bin/start-hbase.sh"
 
-    For more details:
-      https://hbase.apache.org/book.html
+  def plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+    <dict>
+      <key>KeepAlive</key>
+      #{(build.without? "hadoop") ? "<true/>" : "<dict>\n        <key>OtherJobEnabled</key>\n        <string>"+Formula["hadoop"].plist_name+"</string>\n      </dict>"}
+      <key>Label</key>
+      <string>#{plist_name}</string>
+      <key>EnvironmentVariables</key>
+      <dict>
+       <key>HBASE_MASTER_OPTS</key><string> -XX:PermSize=128m -XX:MaxPermSize=128m</string>
+       <key>HBASE_LOG_DIR</key><string>#{var}/hbase</string>
+       <key>HBASE_HOME</key><string>#{opt_libexec}</string>
+       <key>HBASE_SECURITY_LOGGER</key><string>INFO,RFAS</string>
+       <key>HBASE_PID_DIR</key><string>#{var}/run/hbase</string>
+       <key>HBASE_NICENESS</key><string>0</string>
+       <key>HBASE_IDENT_STRING</key><string>root</string>
+       <key>HBASE_REGIONSERVER_OPTS</key><string> -XX:PermSize=128m -XX:MaxPermSize=128m</string>
+       <key>HBASE_OPTS</key><string>-XX:+UseConcMarkSweepGC</string>
+       <key>HBASE_ROOT_LOGGER</key><string>INFO,RFA</string>
+       <key>HBASE_LOG_PREFIX</key><string>hbase-root-master</string>
+       <key>HBASE_LOGFILE</key><string>hbase-root-master.log</string>
+      </dict>
+      <key>ProgramArguments</key>
+      <array>
+        <string>#{opt_bin}/hbase</string>
+        <string>--config</string>
+        <string>#{opt_libexec}/conf</string>
+        <string>master</string>
+        <string>start</string>
+      </array>
+      <key>RunAtLoad</key>
+      <true/>
+      <key>WorkingDirectory</key>
+      <string>#{HOMEBREW_PREFIX}</string>
+      <key>StandardOutPath</key>
+      <string>#{var}/hbase/hbase.log</string>
+      <key>StandardErrorPath</key>
+      <string>#{var}/hbase/hbase.err</string>
+    </dict>
+    </plist>
     EOS
   end
 
   test do
     assert_match /#{version}/, shell_output("#{bin}/hbase mapredcp")
+    system "hbase-start.sh"
+    system "echo stats | nc localhost 2181"
```

Comment:
I used `pipe_output` because that way I can check the output too, whereas if I used plain ruby I'd just wind up reimplementing `pipe_output` anyway.